From 579297be7e2e7f8ae4221033b838de84c152f236 Mon Sep 17 00:00:00 2001
From: "Radu Pirea (NXP OSS)" <radu-nicolae.pirea@oss.nxp.com>
Date: Thu, 20 May 2021 12:28:22 +0300
Subject: [PATCH 08/16] s32g274abluebox3: add qspi support

Add qspi memory node in fsl-s32g274abluebox3.dts and a qspi defconfig.

Issue: ALB-7012

Upstream-Status: Pending

Signed-off-by: Radu Pirea (NXP OSS) <radu-nicolae.pirea@oss.nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/fsl-s32g274abluebox3.dts   | 46 +++++++++++++++++
 configs/s32g274abluebox3_defconfig      |  1 +
 configs/s32g274abluebox3_qspi_defconfig | 68 +++++++++++++++++++++++++
 3 files changed, 115 insertions(+)
 create mode 100644 configs/s32g274abluebox3_qspi_defconfig

diff --git a/arch/arm/dts/fsl-s32g274abluebox3.dts b/arch/arm/dts/fsl-s32g274abluebox3.dts
index 28e807073c..59e7e753ad 100644
--- a/arch/arm/dts/fsl-s32g274abluebox3.dts
+++ b/arch/arm/dts/fsl-s32g274abluebox3.dts
@@ -184,6 +184,34 @@
 				    >;
 		};
 
+		pinctrl0_qspi: pinctrl0_qspi {
+			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
+				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
+				    PF07_MSCR_S32G PF07_QSPI_DATA_A2_CFG
+				    PF08_MSCR_S32G PF08_QSPI_DATA_A3_CFG
+				    PF09_MSCR_S32G PF09_QSPI_DATA_A4_CFG
+				    PF10_MSCR_S32G PF10_QSPI_DATA_A5_CFG
+				    PF11_MSCR_S32G PF11_QSPI_DATA_A6_CFG
+				    PF12_MSCR_S32G PF12_QSPI_DATA_A7_CFG
+				    PF13_MSCR_S32G PF13_QSPI_DQS_A_CFG
+				    PG00_MSCR_S32G PG00_QSPI_CLK_A_CFG
+				    PG01_MSCR_S32G PG01_QSPI_CLK_A_b_CFG
+				    PG02_MSCR_S32G PG02_QSPI_CLK_2A_CFG
+				    PG03_MSCR_S32G PG03_QSPI_CLK_2A_b_CFG
+				    PG04_MSCR_S32G PG04_QSPI_CS_A0
+				    PG05_MSCR_S32G PG05_QSPI_CS_A1
+				    QSPI_DATA0_A_IMCR PF05_QSPI_DATA_A0_IN
+				    QSPI_DATA1_A_IMCR PF06_QSPI_DATA_A1_IN
+				    QSPI_DATA2_A_IMCR PF07_QSPI_DATA_A2_IN
+				    QSPI_DATA3_A_IMCR PF08_QSPI_DATA_A3_IN
+				    QSPI_DATA4_A_IMCR PF09_QSPI_DATA_A4_IN
+				    QSPI_DATA5_A_IMCR PF10_QSPI_DATA_A5_IN
+				    QSPI_DATA6_A_IMCR PF11_QSPI_DATA_A6_IN
+				    QSPI_DATA7_A_IMCR PF12_QSPI_DATA_A7_IN
+				    QSPI_DQS_A_IMCR PF13_DQS_A_IN
+				    >;
+		};
+
 		pinctrl0_sd0: pinctrl0_sd0 {
 			fsl,pins = <PC14_MSCR_S32G PC14_SD0_CLK_CFG
 				    PC15_MSCR_S32G PC15_SD0_CMD_CFG
@@ -262,3 +290,21 @@
 		};
 	};
 };
+
+&qspi {
+	pinctrl-0 = <&pinctrl0_qspi>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	/delete-node/ mx25uw51245g@0;
+
+	mt35xu512aba@0 {
+		address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <200000000>;
+		spi-tx-bus-width = <8>;
+		spi-rx-bus-width = <8>;
+		reg = <0>;
+	};
+};
\ No newline at end of file
diff --git a/configs/s32g274abluebox3_defconfig b/configs/s32g274abluebox3_defconfig
index 7f981dd8c4..7c678c2f5c 100644
--- a/configs/s32g274abluebox3_defconfig
+++ b/configs/s32g274abluebox3_defconfig
@@ -44,6 +44,7 @@ CONFIG_DM_MMC=y
 CONFIG_FSL_USDHC=y
 CONFIG_DM_SPI_FLASH=y
 # CONFIG_SPI_FLASH_BAR is not set
+CONFIG_SPI_FLASH_STMICRO=y
 CONFIG_PHYLIB=y
 CONFIG_PHY_FIXED=y
 CONFIG_DM_ETH=y
diff --git a/configs/s32g274abluebox3_qspi_defconfig b/configs/s32g274abluebox3_qspi_defconfig
new file mode 100644
index 0000000000..422ca645bf
--- /dev/null
+++ b/configs/s32g274abluebox3_qspi_defconfig
@@ -0,0 +1,68 @@
+CONFIG_FLASH_BOOT=y
+CONFIG_ARM=y
+CONFIG_ARCH_S32=y
+CONFIG_ENV_SIZE=0x10000
+CONFIG_ENV_SECT_SIZE=0x1000
+CONFIG_ENV_OFFSET=0x1e0000
+CONFIG_NR_DRAM_BANKS=3
+CONFIG_TARGET_S32G274ABLUEBOX3=y
+CONFIG_S32_LPDDR4=y
+# CONFIG_S32GEN1_DRAM_INLINE_ECC is not set
+CONFIG_DISTRO_DEFAULTS=y
+# CONFIG_SYS_MALLOC_F is not set
+CONFIG_TOOLS_DEBUG=y
+CONFIG_FIT=y
+CONFIG_USE_BOOTARGS=y
+CONFIG_ARCH_EARLY_INIT_R=y
+CONFIG_BOARD_EARLY_INIT_F=y
+CONFIG_BOARD_EARLY_INIT_R=y
+CONFIG_LAST_STAGE_INIT=y
+CONFIG_CMD_EEPROM=y
+CONFIG_CMD_MD5SUM=y
+CONFIG_MD5SUM_VERIFY=y
+CONFIG_CMD_MEMTEST=y
+CONFIG_CMD_CLK=y
+CONFIG_CMD_DM=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_PCI=y
+CONFIG_CMD_SPI=y
+CONFIG_CMD_PMIC=y
+CONFIG_CMD_SMC=y
+CONFIG_DEFAULT_DEVICE_TREE="fsl-s32g274abluebox3"
+CONFIG_ENV_IS_IN_SPI_FLASH=y
+CONFIG_USE_ENV_SPI_BUS=y
+CONFIG_ENV_SPI_BUS=6
+CONFIG_USE_ENV_SPI_CS=y
+CONFIG_ENV_SPI_CS=0
+CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_DM=y
+# CONFIG_DM_DEVICE_REMOVE is not set
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_MXC=y
+CONFIG_MISC=y
+CONFIG_I2C_EEPROM=y
+CONFIG_SYS_EEPROM_SIZE=128
+CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=10
+CONFIG_DM_MMC=y
+CONFIG_FSL_USDHC=y
+CONFIG_DM_SPI_FLASH=y
+# CONFIG_SPI_FLASH_BAR is not set
+CONFIG_SPI_FLASH_STMICRO=y
+CONFIG_PHYLIB=y
+CONFIG_PHY_FIXED=y
+CONFIG_DM_ETH=y
+CONFIG_FSL_PFENG=y
+CONFIG_PHY_GIGE=y
+CONFIG_RGMII=y
+CONFIG_PCI=y
+CONFIG_DM_PCI=y
+CONFIG_PCIE_S32GEN1=y
+CONFIG_PHY=y
+CONFIG_DM_PMIC=y
+CONFIG_DM_PMIC_VR5510=y
+CONFIG_FSL_LINFLEXUART=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_FSL_DSPI=y
+CONFIG_FSL_QSPI=y
-- 
2.17.1

